### PR TITLE
fixes a bug with coupon code redemption

### LINKED ISSUE #215: Coupon codes can not be redeemed

### PR DESCRIPTION
**Pull request by rauhryan**
Closes #215 when merged

### CODE REVIEW

File: vendor/engines/saas/app/controllers/saas/coupons_controller.rb [hunk 1]
```diff
@@ -11,8 +11,8 @@ def valid
     def redeem
       query = Queries::CouchCustomer.get_cust(params[:id], couch)
       doc = QueryHandler.exec(&query)
-      return render json: {success: false, message: "Couldn't find couch record: #{plan_doc.id}"} unless doc
       plan_doc = doc
+      return render json: {success: false, message: "Couldn't find couch record: #{plan_doc.id}"} unless doc
```
Comment (dahlbyk) [inline vendor/engines/saas/app/controllers/saas/coupons_controller.rb]: So if `doc` is `nil`, then `plan_doc` will also be `nil` and `plan_doc.id` will throw...?

Comment (dahlbyk) [inline vendor/engines/saas/app/controllers/saas/coupons_controller.rb]: To be more constructive, it seems like the error message needs to reference `params[:id]`.

Comment (rauhryan) [inline vendor/engines/saas/app/controllers/saas/coupons_controller.rb]: Yep, so I just moved the line down. 

could be refactored to just use doc I suppose

Comment (dahlbyk) [inline vendor/engines/saas/app/controllers/saas/coupons_controller.rb]: But the line is only called `unless doc` - if it's hit, `doc` won't have an `id`.

Comment (discorick) [inline vendor/engines/saas/app/controllers/saas/coupons_controller.rb]: `plan_doc` is `false` if doc fails `unless` -

+1 to "To be more constructive, it seems like the error message needs to reference params[:id]" from @dahlbyk

File: vendor/engines/saas/app/controllers/saas/coupons_controller.rb [hunk 2]
```diff
@@ -11,16 +11,15 @@ def valid
     def redeem
       query = Queries::CouchCustomer.get_cust(params[:id], couch)
       doc = QueryHandler.exec(&query)
-      return render json: {success: false, message: "Couldn't find couch record: #{plan_doc.id}"} unless doc
-      plan_doc = doc
+      return render json: {success: false, message: "Couldn't find couch record: #{doc.id}"} unless doc
```
Comment (dahlbyk) [inline vendor/engines/saas/app/controllers/saas/coupons_controller.rb]: Using `doc` alone now is :cool: but I still think we need to use `params[:id]` for the error message.

**Pull request merged**